extractor/src/main/java/org/schabi/newpipe/extractor/services/youtube/YoutubeThrottlingDecrypter.java

```diff
@@ -97,21 +98,24 @@ public static String apply(final String url, final String videoId) throws Parsin
 
     private static String parseDecodeFunctionName(final String playerJsCode)
             throws Parser.RegexException {
-        String functionName = Parser.matchGroup1(FUNCTION_NAME_PATTERN, playerJsCode);
-        final int arrayStartBrace = functionName.indexOf("[");
-
-        if (arrayStartBrace > 0) {
-            final String arrayVarName = functionName.substring(0, arrayStartBrace);
-            final String order = functionName.substring(
-                    arrayStartBrace + 1, functionName.indexOf("]"));
-            final int arrayNum = Integer.parseInt(order);
-            final Pattern arrayPattern = Pattern.compile(
-                    String.format("var %s=\\[(.+?)\\];", arrayVarName));
-            final String arrayStr = Parser.matchGroup1(arrayPattern, playerJsCode);
-            final String[] names = arrayStr.split(",");
-            functionName = names[arrayNum];
+        final Matcher matcher = FUNCTION_NAME_PATTERN.matcher(playerJsCode);
+        final boolean foundMatch = matcher.find();
+        if (!foundMatch) {
+            throw new Parser.RegexException("Failed to find pattern \""
+                    + FUNCTION_NAME_PATTERN + "\"");
         }
-        return functionName;
+
+        final String functionName = matcher.group(1);
+        if (matcher.groupCount() == 1) {
+            return functionName;
+        }
+
+        final int arrayNum = Integer.parseInt(matcher.group(2));
+        final Pattern arrayPattern = Pattern.compile(
+                "var " + Pattern.quote(functionName) + "\\s*=\\s*\\[(.+?)];");
```

Comment:
```suggestion
                "var " + Pattern.quote(functionName) + "\\s*=\\s*\\[(.+?)][,;]");
```

Otherwise player https://www.youtube.com/s/player/b7910ca8/player_ias.vflset/en_US/base.js breaks